packages/round-manager/src/features/round/RoundDetailForm.tsx

```diff
@@ -234,326 +284,411 @@ export function RoundDetailForm(props: RoundDetailFormProps) {
-                      <svg
-                        xmlns="http://www.w3.org/2000/svg"
-                        className="h-5 w-5"
-                        viewBox="0 0 20 20"
-                        fill="currentColor"
+                  {/* Round end date */}
+                  <div className="col-span-6 sm:col-span-3">
+                    <div
+                      className={`relative border rounded-md px-3 py-2 mb-2 shadow-sm focus-within:ring-1 ${
+                        errors.roundEndTime
+                          ? "border-red-300 text-red-900 placeholder-red-300 focus-within:outline-none focus-within:border-red-500 focus-within: ring-red-500"
+                          : "border-gray-300 focus-within:border-indigo-600 focus-within:ring-indigo-600"
+                      } ${
+                        isRoundEndTimeDisabled
+                          ? "cursor-not-allowed bg-gray-100"
+                          : ""
+                      }`}
+                    >
+                      <label
+                        htmlFor="roundEndTime"
+                        className="block text-[10px]"
                       >
-                        <path
-                          fillRule="evenodd"
-                          d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z"
-                          clipRule="evenodd"
-                        />
-                      </svg>
+                        End Date
+                      </label>
+                      <Controller
+                        control={control}
+                        name="roundEndTime"
+                        render={({ field }) => {
+                          return (
+                            <Datetime
+                              {...field}
+                              closeOnSelect
+                              inputProps={{
+                                id: "roundEndTime",
+                                placeholder: "",
+                                disabled: isRoundEndTimeDisabled,
+                                className:
+                                  "block w-full border-0 p-0 text-gray-900 placeholder-grey-400 focus:ring-0 text-sm disabled:opacity-50 disabled:cursor-not-allowed disabled:bg-gray-100",
+                              }}
+                              onChange={(date) => {
+                                field.onChange(moment(date));
+                                setRoundEndDate(moment(date));
+                                if (rollingApplications) {
+                                  setApplicationEndDate(moment(date));
+                                  setValue(
+                                    "applicationsEndTime",
+                                    moment(date).toDate()
+                                  );
```

Comment:
Can we find a way of letting the reader know:

- what type is `date` in `(date) => {`?
- why are we passing `date` to `moment()` repeatedly? What type is that converting to, if any? 
- why are we converting from that (whatever that is) to `date` before giving it to `setValue`?

I'm mostly concerned that we're making it very easy for the reader to introduce bugs here due to not understanding what's going on.